The LAPACK library of f2cblaslapack depends on the BLAS library in that package.
Link to it if selected.

diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt.orig	2016-06-16 18:24:20.000000000 +0200
+++ b/CMakeLists.txt	2022-01-19 16:37:35.997281700 +0100
@@ -202,6 +202,10 @@
 ## Example use on the command line:
 ## cmake -D OPENBLAS_PROVIDES_LAPACK=true .
 
+option(LAPACK_REQUIRES_BLAS "LAPACK library depends on BLAS library" OFF)
+## Example use on the command line:
+## cmake -D LAPACK_REQUIRES_BLAS=true .
+
 option(ALLOW_FLEXIBLAS_LINUX "Allow detection of FlexiBLAS on Linux" ON)
 # set(ALLOW_FLEXIBLAS_LINUX false)
 ## uncomment the above line to disable the detection of FlexiBLAS;
@@ -384,7 +388,7 @@
       message(STATUS "*** WARNING: found both OpenBLAS and ATLAS; ATLAS will not be used")
     endif()
     
-    if(OpenBLAS_FOUND AND BLAS_FOUND)
+    if(OpenBLAS_FOUND AND BLAS_FOUND AND NOT LAPACK_REQUIRES_BLAS)
       message(STATUS "")
       message(STATUS "*** WARNING: found both OpenBLAS and BLAS; BLAS will not be used")
     endif()
@@ -426,8 +430,17 @@
     if(LAPACK_FOUND)
       set(ARMA_USE_LAPACK true)
       set(ARMA_LIBS ${ARMA_LIBS} ${LAPACK_LIBRARIES})
+      if(LAPACK_REQUIRES_BLAS)
+        if(BLAS_FOUND)
+          set(ARMA_USE_BLAS true)
+          set(ARMA_LIBS ${ARMA_LIBS} ${BLAS_LIBRARIES})
+        else()
+          message(STATUS "")
+          message(STATUS "*** WARNING: LAPACK_REQUIRES_BLAS was set, but no BLAS library could be found")
+        endif()
+      endif()
     endif()
-      
+    
   endif()
   
 endif()
